protocols/secio/src/codec/mod.rs

```diff
@@ -37,6 +39,61 @@ pub type FullCodec<S> = DecoderMiddleware<EncoderMiddleware<length_delimited::Fr
 
 pub type StreamCipher = Box<dyn StreamCipherCore + Send>;
 
+#[derive(Debug, Clone)]
+pub enum Hmac {
+    Sha256(hmac::Hmac<Sha256>),
+    Sha512(hmac::Hmac<Sha512>),
+}
+
+impl Hmac {
+    /// Returns the size of the hash in bytes.
+    #[inline]
+    pub fn num_bytes(&self) -> usize {
+        match *self {
+            Hmac::Sha256(_) => 32,
+            Hmac::Sha512(_) => 64,
+        }
+    }
+
+    /// Builds a `Hmac` from an algorithm and key.
+    pub fn from_key(algorithm: Digest, key: &[u8]) -> Self {
+        // TODO: don't unwrap
+        match algorithm {
+            Digest::Sha256 => Hmac::Sha256(Mac::new_varkey(key).unwrap()),
+            Digest::Sha512 => Hmac::Sha512(Mac::new_varkey(key).unwrap()),
```

Comment:
Maybe replace `unwrap` with `expect` explaining that `HMAC` accepts keys of any size.